### ISSUE TITLE
MD037 false positives with `*` in a `<code>`

### ISSUE THREAD

**Issue by regseb** [DavidAnson/markdownlint#427]
The [_no-space-in-emphasis_](https://github.com/DavidAnson/markdownlint/blob/main/lib/md037.js) rule (_MD037_) displays false positives with `*` in a `<code>` tag. There is no problem with `` ` ``. Maybe the text in `<code>` isn't interpreted as _inCode_.

```markdown
- <code>* *</code> (false positive);
- `* *`.
```

> - <code>* *</code> (false positive);
> - `* *`.

**Comment by DavidAnson** [comment]
MD037 only recognizes Markdown code sections using backtick or indent. HTML tags are discouraged (MD033) and ignored by this rule.

**Comment by regseb** [comment]
I have a big table in my [README](https://github.com/regseb/gout/blob/master/src/module/list/README.md). I use the `<table>` tag to format the code (with line breaks for example). But markdown isn't interpreted in the `<table>` tag, so I use HTML tags (`<code>`, `<a>`...).

```markdown
| Table with backtick |
| -- |
| Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.<br />`* *` (backtick)<br /><code>* *</code> (HTML) |

<table>
  <tr>
    <th>Table with HTML</th>
  </tr>
  <tr>
    <td>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
        dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip
        ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore
        eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia
        deserunt mollit anim id est laborum.
      </p>
      <p>`* *` (backtick)</p>
      <p><code>* *</code> (HTML)</p>
    </td>
  </tr>
</table>
```


> | Table with backtick |
> | -- |
> | Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.<br />`* *` (backtick)<br /><code>* *</code> (HTML) |
> 
> <table>
>   <tr>
>     <th>Table with HTML</th>
>   </tr>
>   <tr>
>     <td>
>       <p>
>         Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
>         dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip
>         ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore
>         eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia
>         deserunt mollit anim id est laborum.
>       </p>
>       <p>`* *` (backtick)</p>
>       <p><code>* *</code> (HTML)</p>
>     </td>
>   </tr>
> </table>

**Comment by DavidAnson** [comment]
I understand what you are doing here, but I am reluctant to add support for HTML tags to the various Markdown rules that have corresponding HTML syntax. By rough approximation it looks like maybe half of the current rules could be applied to HTML as well. It's challenging enough to implement flexible Markdown parsing; doing the same for HTML may be too much. (It's true that I made an exception for MD041, but that was for a very common scenario that's well constrained.)
